lib/tools/apk-utils.js

```diff
@@ -141,26 +141,47 @@ apkUtilsMethods.startApp = async function (startAppOptions = {}) {
  */
 apkUtilsMethods.getFocusedPackageAndActivity = async function () {
   log.debug("Getting focused package and activity");
-  let cmd = ['dumpsys', 'window', 'windows'];
-  let nullRe = new RegExp(/mFocusedApp=null/);
-  let searchRe = new RegExp('mFocusedApp.+Record\\{.*\\s([^\\s\\/\\}]+)' +
-                            '\\/([^\\s\\/\\}\\,]+)\\,?(\\s[^\\s\\/\\}]+)*\\}'); // https://regex101.com/r/xZ8vF7/1
+  const cmd = ['dumpsys', 'window', 'windows'];
+  const nullFocusedAppRe = new RegExp(/mFocusedApp=null/);
+  // https://regex101.com/r/xZ8vF7/1
+  const focusedAppRe = new RegExp('mFocusedApp.+Record\\{.*\\s([^\\s\\/\\}]+)' +
+                                        '\\/([^\\s\\/\\}\\,]+)\\,?(\\s[^\\s\\/\\}]+)*\\}');
+  const nullCurrentFocusRe = new RegExp(/mCurrentFocus=null/);
+  const currentFocusAppRe = new RegExp('mCurrentFocus.+\\{\\w+\\s+([^\\s\\/]+)\\/([^\\s]+)');
+  const getMatch = (pattern, output) => {
```

Comment:
Does this have advantages over a single multiline regular expression?

Reply:
better readability and easier to change IMHO

Reply:
More readable than `m`? I'm not suggesting combining the four regular expressions. I'm just wondering why split the output and run the regular expression a bunch of times. Each regular expression can be run a single time against the entire output.

Reply:
yeah, why not just get the first match in a multiline regex test?

Reply:
got your point. Changed